### PR TITLE
Add zorder option in QuiverKey

### PR DESCRIPTION
**Pull request by muchojp**
## PR Summary
Add zorder option in QuiverKey. So far, if you want to set zorder, you need `set_zorder`. Therefore, I can make `zorder` option available.

## PR Checklist

**Tests and Styling**
- [x] Has pytest style unit tests (and `pytest` passes).
- [x] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).

**Documentation**
- [x] New features are documented, with examples if plot related.
- [N/A] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [x] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).
- [x] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).

### CODE REVIEW

**Comment by oscargus** [comment]
Looks good! If you can please add a test for this it is helpful.

One way may be to use something like:
```
@check_figures_equal()
def test_zorder(fig_test, fig_ref):
    fig_test.quiver(QuiverKey(zorder=...))

   fig_ref.quiver(...)
    ..set_zorder(...)
```
(No idea about the exact syntax, but basically creating the same figure twice: one using the new keyword and one using the old `set_zorder` approach.)

This should go in `tests/test_quiver.py` and you will need to import `check_figure_equal` from `matplotlib.testing.decorators`.

**Comment by jklymak** [comment]
feel free to move out of draft when there is a test.  Feel free to ping if you need more guidance on the test.  Note we already have `test_quiver.py`, which is probably the best place for this.

File: lib/matplotlib/quiver.py [hunk 1]
```diff
@@ -284,6 +285,8 @@ def __init__(self, Q, X, Y, U, label,
             A dictionary with keyword arguments accepted by the
             `~matplotlib.font_manager.FontProperties` initializer:
             *family*, *style*, *variant*, *size*, *weight*.
+        zorder : float or None
+            zorder of the key.
```
Comment (QuLogic) [inline lib/matplotlib/quiver.py:289]:
```suggestion
            The zorder of the key. The default is 0.1 above *Q*.
```

**Comment by muchojp** [comment]
Thank you for a lot of advice. I added a test and new API change doc.
Can you review this PR?

File: lib/matplotlib/tests/test_quiver.py [hunk 2]
```diff
@@ -277,3 +278,19 @@ def test_quiver_setuvc_numbers():
 
     q = ax.quiver(X, Y, U, V)
     q.set_UVC(0, 1)
+
+
+@check_figures_equal()
```
Comment (QuLogic) [inline lib/matplotlib/tests/test_quiver.py:283]: Probably no need to check other types:
```suggestion
@check_figures_equal(extensions=['png'])
```

File: lib/matplotlib/tests/test_quiver.py [hunk 3]
```diff
@@ -277,3 +278,19 @@ def test_quiver_setuvc_numbers():
...
+def test_zorder(fig_test, fig_ref):
+    """Check QuiverKey zorder option"""
+    X, Y, U, V = 1, 1, 2, 2
+
+    ax_test = fig_test.subplots()
+    zorder_value = 5.
```
Comment (QuLogic) [inline lib/matplotlib/tests/test_quiver.py:289]: The default `zorder` for `quiver` is 1, so the default for `quiverkey` is 1.1. That means `zorder=5` may effectively be the same as not specifying anything? So I think this test needs at least another artist in between 1.1 and 5 to confirm that the key really is moved up.

**Comment by jklymak** [comment]
Moved back to draft until the test can be fixed.  Thanks!

**Comment by rcomer** [comment]
Hi @muchojp are you still interested in working on this?

**Pull request closed without merging**